melloddy_predictor/predictor_single.py

```diff
@@ -0,0 +1,411 @@
+        self.device = device
+        self.net = sc.SparseFFN(self.conf).to(self.device)
+        self.inverse_normalization = False
+        state_dict = torch.load(model, map_location=torch.device(self.device))
+
+        if self.conf.model_type == "federated":
+            state_dict_new = OrderedDict()
+            state_dict_new["net.0.net_freq.weight"] = state_dict["0.0.net_freq.weight"]
+            state_dict_new["net.0.net_freq.bias"]   = state_dict["0.0.net_freq.bias"]
+            state_dict_new["net.2.net.2.weight"]    = state_dict["1.net.2.weight"]
+            state_dict_new["net.2.net.2.bias"]      = state_dict["1.net.2.bias"]
+            state_dict = state_dict_new
+
+        #apply model weights
+        self.net.load_state_dict(state_dict)
+        #set model into evaluation mode
+        self.net.eval()
+        #apply dropout, if chosen
+        self.dropout = dropout
+        if self.dropout:
+            self.net.apply(sc.utils.enable_dropout) 
+        # if inverse normalization is done load the stats
+        if 'stats' in results_loaded:
+            self.inverse_normalization = True
+            stats = results_loaded["stats"]
+            self.reg_mean = np.array(stats["mean"])
+            self.reg_var = np.array(stats["var"])
+            self.reg_stddev = np.sqrt(self.reg_var)
+        
+        if self.net.cat_id_size is not None:
+            raise NotImplementedError("Predictions for models with a catalog head are not yet implemented")
+        if self.net.class_output_size > 0:
+            if self.net.regr_output_size > 0:
+                self.model_type = ScModelType.hybrid
+            else:
+                self.model_type = ScModelType.classification
+        elif self.net.regr_output_size > 0:
+            self.model_type = ScModelType.regression
+
+        self.has_task_maps = False
+        if (class_task_map is not None) or (regr_task_map is not None):
+            self.set_tasks_maps(class_task_map, regr_task_map)
+        
+
+    def set_tasks_maps(self, class_task_map=None, regr_task_map=None):
+        """
+        Set the task maps stored in the object
+        
+        Args:
+            class_task_map:              a dictionary or pandas series having classification task labels as as key or index, resp, and continuous classification task IDs (column indexes of the prediction matrix) as values
+            regr_task_map:               a dictionary or pandas series having regression task labels as as key or index, resp, and continuous regression task IDs (column indexes of the prediction matrix) as values 
+        
+        """
+        class_task_map, regr_task_map, mapped_tasks_type = self.__validate_maps__(class_task_map, regr_task_map)
+        self.class_task_map = class_task_map
+        self.regr_task_map = regr_task_map
+        self.mapped_tasks_type = mapped_tasks_type
+        self.has_task_maps = True
+
+
+    def get_mapped_task_names(self):
```

Comment:
For the functions not used/tested, could you add a test or remove them ?

Reply:
test were added